compiler/rustc_typeck/src/expr_use_visitor.rs

```diff
@@ -228,8 +233,65 @@ impl<'a, 'tcx> ExprUseVisitor<'a, 'tcx> {
             }
 
             hir::ExprKind::Match(ref discr, arms, _) => {
+                // ROX: Introduce a borrow here if the pattern needs to reads the place
+                //      - If the value of the expression will be ever read
+                //      - i.e. it contains more than just a `_` being matched on
+                fn check_for_no_wild_card(pat: &Pat<'hir>) -> bool {
```

Comment:
For later but I don't think you need to do this manually. 

See walk_pat
```
        let tcx = self.tcx();
        let ExprUseVisitor { ref mc, body_owner: _, ref mut delegate } = *self;
        return_if_err!(mc.cat_pattern(discr_place.clone(), pat, |place, pat| {
```

If the pat isn't a wild then you're good. I think this would work or something similar.